portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,94 @@
 
 import java.io.IOException;
+import javax.servlet.*;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.ArrayList;
+
+import com.google.appengine.api.datastore.*;
+import java.io.*;
+import java.util.*;
+import com.google.appengine.api.datastore.Query.SortDirection;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List <String> comments = new ArrayList();
+
+  @Override
+  public void init() throws ServletException {
+        comments.add("Hello");
+        comments.add("My");
+        comments.add("name");
+        comments.add("is");
+        comments.add("Kearsten");
+        comments.add("Kearsten");
+    }
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello kip!</h1>");
+    response.setContentType("application/json;");
+    response.getWriter().println(comments);
   }
+  private String commentsToJson(List<Comment> comments){
+      Gson gson = new Gson();
+      return gson.toJson(comments);
+  }
+  private List<Comment> getcomments(){
+      Query query = new Query("Comment").addSort("time", SortDirection.DESCENDING);
+      List <Comment> comments = new ArrayList();
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+      PreparedQuery results = datastore.prepare(query);
+
+      /* for(Entity entity: results.asIterable())
+      {
+          String text = (String) entity.getProperty("text");
+          long time = (long) entity.getProperty("time");
+          comments.add(new Comment(text,time));
+      } */
+      return comments;
+      }
+  /*private List <String> getCommentsMock(){
+        List <String> comments = new ArrayList();
+        comments.add("Hello");
+        comments.add("My");
+        comments.add("name");
+        comments.add("is");
+        comments.add("Kearsten");
+        return comments;
+      }*/
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String text = request.getParameter("comment_post_ID");
+    if(text == null || text.equals("")){
+        return;
+    }
+
+    addComment(text, System.currentTimeMillis());
+    response.sendRedirect("/index.html");
+  }
+    private void addComment(String commentText, long time){
+        Entity commentEntity = new Entity("comment");
+        commentEntity.setProperty("text", commentText);
+        commentEntity.setProperty("time", time);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
```

Comment:
When you begin updating the comments, try working with the "comments" class member first instead of datastore. You can simply add strings like in your init method. Once that is working, then you know that your handling of gets and posts is correct and you can more easily add datastore to your app